feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -74,6 +78,59 @@ import kotlinx.coroutines.delay
 private const val TAG = "PreviewScreen"
 private const val BLINK_TIME = 600L
 
+@Composable
+fun AmplitudeVisualizer(
+    modifier: Modifier = Modifier,
+    sizeScale: Int = 100,
+    audioAmplitude: Double
+) {
+    val animatedSize by animateDpAsState(
+        targetValue = (
+            (sizeScale * 1.5)
+                .coerceAtMost(
+                    sizeScale *
+                        (
+                            1 + EaseOutExpo.transform(audioAmplitude.toFloat())
+                                .toDouble()
+                            )
+                )
+            ).dp,
+        label = "AudioAnimation"
+    )
+    Box(modifier = modifier) {
+        Canvas(
+            modifier = Modifier
+                .size(animatedSize)
```

Comment:
Instead of updating the size here, could you use a fixed max size and draw a circle of the needed size inside the Canvas onDraw? That way you won't need a new layout pass every time the value change (which will be every frame I guess), Compose will be smart enough to only re-draw.

Reply:
Good idea! it hadn't crossed my mind